### PR TITLE
Ch- setup- docker- 26

### PR DESCRIPTION
**Pull request by faid-terence**
### What does the PR do?
Sets-up and created Dockerfile
I've created a Dockerfile for a Node.js application and I would like to contribute it to the project. This Dockerfile is based on the official Node.js image and includes all the necessary steps to build and run a Node.js application.
#### Description of Task to be completed?
1.An image of the backend application has been created
2.Database is running in a docker container
3.Both docker images can communicate while up and running locally
#### How should this be manually tested?
1.It starts with the official Node.js image as the base image.
2.It sets the working directory to /app and copies the package.json and package-lock.json files to the container.
3.It runs npm install to install the dependencies.
4.It copies the rest of the application files to the container.
5.It exposes port 3000 and sets the command to run the application using the npm start script.

### To build and run the container, you can use the following commands:
` docker-compose -f docker-compose.yml -f docker-compose.dev.yml  up -d --build
`
#### What are the relevant pivotal tracker/Trello stories?
CardID: #26
cardLink: https://trello.com/c/cMwZJMBa/26-setup-docker
#### Screenshots (if appropriate)
Screenshot 2023-03-31 at 11 02 53
Screenshot 2023-03-31 at 11 03 34
Screenshot 2023-03-31 at 11 07 54

### CODE REVIEW

File: routes/welcome.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Imports
+import { Router } from 'express';
+
+// Initialization
+const router = Router();
+
+router.use((req, res) => {
+  if (req.accepts('html')) {
+    res.status(200).send('Welcome to the Ecommerce RESTful API!');
```
Comment (ALVINdimpos) [inline routes/welcome.js]: great job faid

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# Copy the current directory contents into the container at /app
+COPY . /app
+
+# Install any needed packages specified in package.json
+RUN npm install
+
+# Make port 3000 available to the world outside this container
+EXPOSE 3000
+
+# Define the command to run the application
```
Comment (chanceDira) [inline Dockerfile]: Let's use "npm start"

**Review by chanceDira — CHANGES_REQUESTED** [review]
Good progress @faid-terence, where is the docker-compose file?

**Comment by faid-terence** [comment]
> Good progress @faid-terence, where is the docker-compose file?

Thanks @chanceDira , I have drafted a docker composé file but I haven't pushed it because we're waiting for database credentials from a person working on it , Hopefully by today it will done 
Thanks

File: Dockerfile (hunk 2 above)
Comment (faid-terence) [reply Dockerfile]: > Let's use "npm start"

Alright it's Okay

**Comment by yobuir** [comment]
> Good progress @faid-terence, where is the docker-compose file?

As @faid-terence mentioned we are working on the project image and we have the setup of `docker-compose` file which contains two images one for project and other for database

**Comment by faid-terence** [comment]
[Setup Docker](https://trello.com/c/cMwZJMBa/26-setup-docker)

**Review by yobuir — APPROVED** [review]

**Review by ALVINdimpos — APPROVED** [review]

**Review by ALVINdimpos — APPROVED** [review]

**Review by bryan936M — APPROVED** [review]

**Review by chanceDira — APPROVED** [review]
Good job @faid-terence @yobuir

**Pull request merged**